### ISSUE TITLE
[Accessibility][Keyboard navigation] Picker does not announce when item is removed from selected list using backspace key.

### ISSUE THREAD

**Issue by alexwaibel** [microsoft/fluentui#16097]
### Environment Information

- **Package version(s)**: @fluentui/react 7.129.0
- **Browser and OS versions**: Windows 10, Edge Version 86.0.622.69

### Describe the issue:

When a user uses the backspace key to remove a previously selected item from the picker, aria does not announce that the item was removed. Instead it just reads through the list of selected results (which could be empty).

Seems that this case was missed in  #11022

### Please provide a reproduction of the issue in a codepen:

You can see this behavior by enabling a screen reader and using the "People Picker with Preselected Items" example from [Fluent's own picker documentation](https://fabricweb.z5.web.core.windows.net/oufr/7.153.2/#/examples/peoplepicker).

#### Actual behavior:

Screen reader says nothing when an item is deleted using the backspace key and instead reads the list of selected individuals. If there was previously only one selected individual, and that individual is removed with a backspace, the list is then empty and the screen reader says nothing at all.

#### Expected behavior:

Screen reader should announce that the item was removed.

### Documentation describing expected behavior

The [aria-relevant](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-relevant_attribute) directive "all" may be useful here.

**Comment by JustSlone** [comment]
@alexwaibel thanks for bringing this issue up, we actually have a super old issue tracking this over here: #11467 looks like it hit a stall implementing it back then. 

Thanks for the tip on aria-relevant, however, I couldn't get aria-relevant to result in announcing the removal we'll have to investigate more. 

As a fallback, we could probably use an Announce component to add an alert announcing the removal. Would be better if we can get aria-relevant to work though.

**Comment by brucehuy** [comment]
@alexwaibel / @JustSlone - any update on this bug or #11467? This is impacting a11y compliance in our service, hoping we can make some progress on it.
